test/suites/kubernetesupgrade/suite_test.go

```diff
@@ -0,0 +1,118 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetesupgrade_test
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1alpha2"
+	"github.com/Azure/karpenter-provider-azure/test/pkg/environment/azure"
+
+	karpv1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+
+	coretest "sigs.k8s.io/karpenter/pkg/test"
+)
+
+var env *azure.Environment
+var nodeClass *v1alpha2.AKSNodeClass
+var nodePool *karpv1.NodePool
+
+func TestDrift(t *testing.T) {
+	RegisterFailHandler(Fail)
+	BeforeSuite(func() {
+		env = azure.NewEnvironment(t)
+	})
+	RunSpecs(t, "KubernetesUpgrade")
+}
+
+var _ = BeforeEach(func() {
+	env.BeforeEach()
+	nodeClass = env.DefaultAKSNodeClass()
+	nodePool = env.DefaultNodePool(nodeClass)
+})
+var _ = AfterEach(func() { env.Cleanup() })
+var _ = AfterEach(func() { env.AfterEach() })
+
+var _ = Describe("KubernetesUpgrade", func() {
+	// NOTE: This Describe has only one test, and no BeforeEach, as this test preforms modifications to the managed cluster in a way that can not be undone,
+	// meaning the environment can not be reset for any additional tests.
+	//
+	// This is actually a test of Drift. However, testing kubernetes upgrade has to be isolated due to its modification of the managed cluster environment.
+	It("should drift and replace all nodes when the cluster's kubernetes version upgrades", func() {
+		numPods := 1
+		// Add pods with a do-not-disrupt annotation so that we can check node metadata before we disrupt
+		dep := coretest.Deployment(coretest.DeploymentOptions{
+			Replicas: int32(numPods),
+			PodOptions: coretest.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "my-app",
+					},
+					Annotations: map[string]string{
+						karpv1.DoNotDisruptAnnotationKey: "true",
+					},
+				},
+				TerminationGracePeriodSeconds: lo.ToPtr[int64](0),
+			},
+		})
+		selector := labels.SelectorFromSet(dep.Spec.Selector.MatchLabels)
+
+		env.ExpectCreated(dep, nodeClass, nodePool)
+		pod := env.EventuallyExpectHealthyPodCount(selector, numPods)[0]
+		nodeClaim := env.EventuallyExpectCreatedNodeClaimCount("==", 1)[0]
+		node := env.ExpectCreatedNodeCount("==", 1)[0]
+
+		By("getting latest available kubernetes version upgrade")
+		availableKubernetesUpgrades := env.ExpectSuccessfulGetOfAvailableKubernetesVersionUpgradesForManagedCluster()
+		sort.Slice(availableKubernetesUpgrades, func(i, j int) bool {
+			return *availableKubernetesUpgrades[i].KubernetesVersion < *availableKubernetesUpgrades[j].KubernetesVersion
```

Comment:
Do we need to care about `IsPrevew`?

Reply:
I was considering this, but felt like testing with the latest was probably best, regardless of IsPreview. Should get use new version tests as they're released, which is nice.